wfl/fit/mace.py

```diff
@@ -0,0 +1,205 @@
+        sys.stderr.write(stderr)
+    
+        # no outputs to rename since everything should be in run_dir
+        xpr.mark_processed()
+        
+        return results
+    
+    if not run_dir.exists():
+        run_dir.mkdir(parents=True)
+    
+    if isinstance(mace_fit_params, str):
+        mace_fit_params = yaml.safe_load(Path(mace_fit_params).read_text())
+    elif isinstance(mace_fit_params, dict):
+        pass
+    
+    for key, val in mace_fit_params.items():
+        if isinstance(val, int) or isinstance(val, float):
+            mace_fit_cmd += f" --{key}={val}"
+        elif isinstance(val, str):
+            mace_fit_cmd += f" --{key}='{val}'"
+        elif val is None:
+            mace_fit_cmd += f" --{key}"
+        else:
+            mace_fit_cmd += f" --{key}='{val}'"
+   
+ 
+    if not do_fit or verbose:
+        print('fitting command:\n', mace_fit_cmd)
+    
+    # Not totally sure if this part is applicable to MACE cpu training.   
+    orig_omp_n = os.environ.get('OMP_NUM_THREADS', None)
+    if 'WFL_GAP_FIT_OMP_NUM_THREADS' in os.environ:
+        os.environ['OMP_NUM_THREADS'] = os.environ['WFL_GAP_FIT_OMP_NUM_THREADS']
+    
+    try:
+        remote_cwd = os.getcwd()    
+        if str(run_dir) != ".":
+            for input_file in kwargs["input_files"]:
+                file_name = input_file.split("/")[-1]
+                
+                # If initiated with previous checkpoint file, it should be copied to 
+                # current fitted directory after creating checkpoint folder. 
+                if file_name.endswith("_swa.pt"):
+                    checkpoint_dir = Path(run_dir / 'checkpoints')  
+    
+                    checkpoint_dir.mkdir(parents=True, exist_ok=True)
+                    copyfile(input_file, f"{checkpoint_dir}/{file_name}")
+    
+                else:
+                    copyfile(input_file, f"{run_dir}/{file_name}")
+    
+            os.chdir(run_dir)
+            subprocess.run(mace_fit_cmd, shell=True, check=True)
+            os.chdir(remote_cwd)    
+        else:
+            subprocess.run(mace_fit_cmd, shell=True, check=True)
+    
+    except subprocess.CalledProcessError as e:
+        print("Failure in calling MACE fitting with error code:", e.returncode)
+        raise e
+
+
+
+def _write_fitting_configs(fitting_configs, use_params, mace_file_base):
+    """
+    Writes fitting configs to file and updates MACE fitting parameters.
+    Configurations and filename handled by Workflow overwrite any filename
+    specified in parameters.
+
+    Parameters:
+    -----------
+    fitting_configs: list(Atoms)
+        configurations to fit to
+    use_params: dict
+        ACE fit parameters, will have input filename set based on where configs were written to
+    ace_file_base: str
+        base to all ACE-related files, used for saving fitting configs
+    """
+
+    assert isinstance(fitting_configs, list)
+
+    if "train_file" not in use_params:
```

Comment:
For this wrapper to make sense in the context of the rest of wfl I think we should assume that the configs are received as a `ConfigSet`, and not support a parameter field that specifies a file. (see how `ace/fit.py` does it). Also, why are you initializing it to an empty dict? Later you set it to a string.

Reply:
Assuming two initialization situations: 1) List of ASE objects 2) xyz file. 
In case initiated with list of ASE object, it writes xyz file when fitting, but requires `fit_cfgs_fname` as argument. 
If initiated from xyz file, then it's copied to `run_dir`. 

Also, I wanted to have situation where user wants to restart from previous checkpoint file, then it should be copied by `remote_info.input_files` and `prev_checkpoint_file` argument should be provided.

Reply:
One of the points of `wfl` is the abstraction of input lists of configs. Please take a look at how the other fitting codes deal with the input coming in as a `ConfigSet`, which could be a file, or many files, or a list of configs, or in principle even something else (e.g. a query to a database that returns a particular set of configs you can iterate over).

Reply:
Note that checkpointing is a separate can of worms, because it isn't a list of configs that wfl has specialized infrastructure for. I'll take a look at the revised code.

Reply:
I understood your concerns regarding `ConfigSet` and largely borrowed some parts from `GAP/simple.py`.

Reply:
Looks good.  See my little remaining suggestion about refactoring that code.